### PR TITLE
Pass MF2WG number test

### PR DESCRIPTION
**Pull request by mvilks**
Lex number literal at the end of lexName()

### CODE REVIEW

File: parse/lex.go [hunk 1]
```diff
@@ -465,14 +469,20 @@ func lexName(l *lexer) stateFn {
 		r rune   // current rune
 	)
 
-	for r = l.next(); isName(r); r = l.next() {
+	for r = l.next(); isName(r) || r == '+'; r = l.next() {
```
Comment (janishorsts) [inline parse/lex.go:472]: is `+` not allowed in name?

Comment (janishorsts) [reply parse/lex.go:472]: If it is not... does it not conflict with lexName() intent?

Comment (mvilks) [reply parse/lex.go:472]: starting char is: ALPHA / "_" / %xC0--> up with some exclusions
name char is: starting char / DIGIT / "-" / "." / %xB7 and something above 0x300
`+` is 0x2B so doesn't fit in any of previous allowed ranges

Comment (janishorsts) [reply parse/lex.go:472]: Does it imply we have to rename `lexName` to `lexNameOrNumber`?

Comment (janishorsts) [reply parse/lex.go:472]: Also. If `s` contains `+` and is NOT `number`, it is NOT `name`. Additional check required.

**Review by janishorsts — APPROVED** [review]

**Pull request merged**